connect/api/src/test/java/org/apache/kafka/connect/data/ValuesTest.java

```diff
@@ -67,6 +70,122 @@
         INT_LIST.add(-987654321);
     }
 
+    @Test
+    public void shouldNotParseUnquotedEmbeddedMapKeysAsStrings() {
```

Comment:
Unless i'm misunderstanding something, I think these test names are inverted.
```suggestion
    public void shouldParseUnquotedEmbeddedMapKeysAsStrings() {
```

Reply:
The "as strings" part refers to "unquoted embedded map keys", as opposed to "maps with unquoted embedded keys". If that's unclear, maybe just `shouldNotParseUnquotedEmbeddedMapKeys` or `shouldFailToParseUnquotedEmbeddedMapKeys`?

Reply:
I read these as "should not parse X as Y", as in X is in the input, and a Y is the output.
The input was a string which included an "unquoted embedded map key", and the output was a string, rather than a fully-parsed map.
I now see the reading you're using, and it makes sense, since it refers to the innermost element that's being tested, rather than the whole input object.